### PR TITLE
Adding vuid 04562

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -4272,9 +4272,27 @@ bool CoreChecks::PreCallValidateCmdBlitImage(VkCommandBuffer commandBuffer, VkIm
 }
 
 bool CoreChecks::PreCallValidateCmdBlitImage2KHR(VkCommandBuffer commandBuffer, const VkBlitImageInfo2KHR *pBlitImageInfo) const {
-    return ValidateCmdBlitImage(commandBuffer, pBlitImageInfo->srcImage, pBlitImageInfo->srcImageLayout, pBlitImageInfo->dstImage,
-                                pBlitImageInfo->dstImageLayout, pBlitImageInfo->regionCount, pBlitImageInfo->pRegions,
-                                pBlitImageInfo->filter, COPY_COMMAND_VERSION_2);
+    bool skip = false;
+
+    auto src_image_state = GetImageState(pBlitImageInfo->srcImage);
```
Comment (ncesario-lunarg) [inline layers/buffer_validation.cpp:4277]: `Get<IMAGE_STATE>`?

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 2]
```diff
@@ -14014,3 +14014,85 @@ TEST_F(VkLayerTest, SparseImageMemoryBindOffset) {
...
+                                                  &copy_command_transform,
+                                                  src_image.image(),
+                                                  src_image.Layout(),
+                                                  dst_image.image(),
+                                                  dst_image.Layout(),
+                                                  1,
+                                                  &blitRegion2,
+                                                  VK_FILTER_NEAREST};
+    // Unsigned int vs not an int
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkBlitImageInfo2KHR-pNext-pNext");
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:14093]: Is `04562` being tested somewhere else?

**Review by ncesario-lunarg — COMMENTED** [review]
Not sure if I'm understanding the test correctly, but the layer code LGTM!

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 3]
```diff
@@ -14034,6 +14034,89 @@ TEST_F(VkLayerTest, SparseImageMemoryBindOffset) {
     m_errorMonitor->VerifyFound();
 }
 
+TEST_F(VkLayerTest, BlitImageWithCopyCommandTransform) {
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:14037]: Looks like this is failing on the mock ICD: https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2997/checks?check_run_id=3134769607#step:5:3384.

Also, is the multi-planar format case tested here?

Comment (ziga-lunarg) [reply tests/vklayertests_buffer_image_memory_sampler.cpp:14037]: No it is not, I will add a test for that too

**Pull request closed without merging**